### ISSUE TITLE
`$modelUpdater` never turns off

### ISSUE THREAD

**Issue by TheSharpieOne** [janpantel/angular-sails#52]
[`$modelUpdater`](https://github.com/janpantel/angular-sails/blob/master/src/service/angular-sails.js#L231) calls `on()` listening for updates and updating the model when an update comes in.  But the listener never turns off.  When the user navigates and the controller and scope is destroyed, the `on` listener is still there.

I would be nice to provide a way to turn off the `$modelUpdater` manually as well as have it automatically turn off when a scope is destroyed.

**Comment by TheSharpieOne** [comment]
This is open for discussion as to how this should work.  @psi-4ward since you created the feature, I would like your input.

My thoughts as to how this could work.

First, create a function that when called removed the listener that was created.  Return the function to the call.  Much like angular's `$watch()` returns a function that when called removes the `$watch`. `$modelUpdater()` will return a function that when called removes the `$modelUpdater()`.  This change would make `$modelupdater` no longer chainable.

Second, to have the listener automatically removed, maybe have a third argument be optionally passed to `$modelUpdater()`: `$modelUpdater(name, models[, scope])`.  Then we can do `scope.on('$destory', removeListener)` in the `$modelUpdater` where `removeListener` is the function noted above (that would be returned in the initial call).  The downside to this is I don't think passing around `scope` is a great idea...  But this would prevent devs from having to manually bind in the controller.

**Comment by psi-4ward** [comment]
Youre right, we should return a removeListener function cause its the most angular-like way.

```javascript
$scope.$on('$destroy', $sails('myModel', $scope.arrModels));
```
looks not so bad doenst it?

PS: sorry for my absence, im really busy atm and we use loopback for our current project

**Comment by TheSharpieOne** [comment]
Yeah, thats not too bad.

**Comment by robgrimball** [comment]
This would be really great to have, to be able to destroy this listener, since it essentially causes a memory leak while switching paths in Angular.  If I have a large list that I am passing to $modelUpdater each time a controller is instantiated, and then wander off to another path, the $scope for that controller goes away, but the list that was passed to $modelUpdater remains until the listener is deleted (ie, presently: never).  This really depletes memory usage rapidly in a single page application wherein the user might be bouncing back and forth between many menu options that are being monitored real time many times.  Though in a non-SPA the issue is moot.  Is there an easy workaround to this in the mean time that I can use?

Ie, if I do : $sails.$modelUpdater('user', userList); is there a way to end up deleting the userList that the listener has or better yet the listener and userList both?

**Comment by TheSharpieOne** [comment]
@robgrimball Right now there is no work-around, but since this is a simple fix. I can push something up to the current release branch.

**Comment by robgrimball** [comment]
That would be great if you could push it up.  I will grab it as soon as it is available, thanks!  Very nice module btw!

**Comment by TheSharpieOne** [comment]
I pushed a new release that fixes this.  See #57 for how to use it.  I'm going to leave this open as this change will need to be made in 2.0

**Comment by TheSharpieOne** [comment]
Resolved with #57 (1.x) and #69 (2.x)
